### PR TITLE
TST: split tests for windows to sub-modules

### LINKED ISSUE #19228: TST: split tests for windows to sub-modules
after #18723 

can split test_window.py to a sub-dir ``pandas/tests/window/`` and have
``test_rolling, expanding, emwa`` and prob ``test_window`` (for other)

### PR DESCRIPTION
**Pull request by giba0**
Splitting the test_window.py file into the test_rolling.py files, test_expanding.py
and test_ewma.py

- [x] closes #19228
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
looks like you picked up some other commits (in the parquet module), can you merge master (or revert these if you did those).

**Comment by jreback** [comment]
want to parameterize some of these, but can wait for another pass for that.

**Comment by gilbertoolimpio** [comment]
@jreback I'm a little confused, because I sent two PR's, but I did not understand why it was all together! Can you help me?

**Comment by jreback** [comment]
for a new PR checkout from master like this
git checkout -b new_branch origin/master

for now you need to rebase on master and remove those commits - like

git rebase -i origin/master

and delete the uneeded commits

**Comment by gilbertoolimpio** [comment]
@jreback I did the suggested fix, could you check it now?

**Comment by jreback** [comment]
someting when really wrong. you need to ``git merge upstream/master`` (if ``upstream`` is the pandas-dev repo, sometimes also called ``origin``).

only *your* changes should be here.

**Comment by gilbertoolimpio** [comment]
Now I think it worked! Could you check, please?

**Comment by jreback** [comment]
you have a ton of other changes in here. you need to have just the 4 files (the original and the split off ones)

File: pandas/tests/windows/test_rolling.py [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+import pandas.core.window as rwindow
+from pandas.errors import UnsupportedFunctionCall
+import pandas.util.testing as tm
+import pandas.util._test_decorators as td
+from pandas.compat import range
+
+N, K = 100, 10
+
+
+def assert_equal(left, right):
```
Comment (jreback) [inline pandas/tests/windows/test_rolling.py]: don't repeat things. 2 ways to solve this, have a common.py and import, or you can use fixtures to do this properly.

File: pandas/tests/windows/test_expanding.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            c(min_periods=1, center=False)
+
+            # not valid
+            for w in [2., 'foo', np.array([2])]:
+                with pytest.raises(ValueError):
+                    c(min_periods=w)
+                with pytest.raises(ValueError):
+                    c(min_periods=1, center=w)
+
+    def test_numpy_compat(self):
```
Comment (jreback) [inline pandas/tests/windows/test_expanding.py]: parameterize this

File: pandas/tests/windows/test_expanding.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    def test_doc_string(self):
+
+        df = DataFrame({'B': [0, 1, 2, np.nan, 4]})
+        df
+        df.expanding(2).sum()
+
+    def test_constructor(self):
+        # GH 12669
+
+        for o in [self.series, self.frame]:
```
Comment (jreback) [inline pandas/tests/windows/test_expanding.py]: create a fixture for series and frame (in conftest.py)

File: pandas/tests/windows/test_ewma.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def assert_equal(left, right):
+    if isinstance(left, Series):
+        tm.assert_series_equal(left, right)
+    else:
+        tm.assert_frame_equal(left, right)
+
+
+class Base(object):
```
Comment (jreback) [inline pandas/tests/windows/test_ewma.py]: see below

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you update

File: pandas/tests/windows/test_expanding.py (hunk 2 above)
Comment (gilbertoolimpio) [reply pandas/tests/windows/test_expanding.py]: Can you hlep me? Here I need use a decorator `@pytest.mark.parametrize`?

File: pandas/tests/windows/test_rolling.py (hunk 1 above)
Comment (gilbertoolimpio) [reply pandas/tests/windows/test_rolling.py]: In this case I just need to insert `@pytest.fixture`?

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**